### PR TITLE
Improvements to gostacktool

### PR DESCRIPTION
**Pull request by kwalkerz**
Mostly addition of 'monitor' feature, which can collect counts over time (either triggered manually or at regular intervals).

### CODE REVIEW

**Comment by kwalkerz** [comment]
retest this

**Comment by kwalkerz** [comment]
retest this

File: tools/gostacktool/gostack/stacktrace.py [hunk 1]
```diff
@@ -70,3 +77,139 @@ def __str__(self):
             string += ''.join(goroutineStrings[1:])
             string += '\n'
         return string
+
+
+# Multiple data structures to monitor the number of Goroutines over time.
+class MonitoredStackTrace(object):
+    def __init__(self, directory):
+        if directory != None and len(directory) > 0:
```
Comment (smousa) [inline tools/gostacktool/gostack/stacktrace.py]: directory != None or directory is not None

Comment (kwalkerz) [inline tools/gostacktool/gostack/stacktrace.py]: Yes, 'is not' is more correct.  I'll update those.

**Comment by kwalkerz** [comment]
retest this

**Comment by kwalkerz** [comment]
retest this

**Comment by kwalkerz** [comment]
retest this

I've just cleared the docker images from the test slave that keeps failing--hopefully that will help...

**Comment by kwalkerz** [comment]
retest this

**Pull request merged**